src/routing/kad-dht.md

```diff
@@ -0,0 +1,719 @@
+efficient and rapid navigation of the network during lookups.
+
+### Bucket Size
+
+The Routing Table MUST contain information about at least `k` DHT Servers whose
+Kademlia Identifier shares a common prefix of length `l` with the local node,
+for every `l` in `[0, 255]`, provided such nodes exist. The set of `k` peers
+sharing a common prefix of length `l` with the local node is called the
+_bucket_ `l`.
+
+In practice, buckets with smaller indices will typically be full, as many nodes
+in the network share shorter prefix lengths with the local node. Conversely,
+buckets beyond a certain index usually remain empty, since it's statistically
+unlikely that any node will have an identifier sharing a very long common
+prefix with the local node. For more information see [bucket population
+measurements](https://github.com/probe-lab/network-measurements/blob/master/results/rfm19-dht-routing-table-health.md#peers-distribution-in-the-k-buckets).
+
+The IPFS Kademlia DHT uses a bucket size of `k = 20`. This corresponds to the
+`k` value as defined in the original Kademlia paper [`[0]`](#bibliography). The `k` value is also
+used as a replication factor and defines how many peers are returned to a
+lookup request.
+
+Note that DHT Clients are never included in a Routing Table.
+
+Each DHT Server MUST store the public
+[multiaddresses](https://github.com/libp2p/specs/blob/master/addressing/README.md)
+for every node in its Routing Table. DHT Servers MUST discard nodes with only
+private and/or relay multiaddresses. Additionally, DHT Servers MUST verify that
+these nodes are reachable and replace any nodes that are no longer accessible.
+
+### Replacement Policy
+
+Nodes MUST NOT be removed from the Routing Table as long as they remain online.
+Therefore, the bucket replacement policy is based on seniority, ensuring that
+the most stable peers are eventually retained in the Routing Table.
+
+#### IP Diversity Filter
+
+DHT servers SHOULD implement an IP Diversity Filter to ensure that nodes in
+their routing table originate from a diverse set of Autonomous System Numbers
+(ASNs). This measure helps mitigate Sybil attacks and enhances the network’s
+resilience.
+
+A recommended approach is to impose the following limits:
+
+* **Globally**, a maximum of `3` nodes sharing the same IP grouping should be
+allowed in the routing table.
+* **Per routing table bucket**, a maximum of `2` nodes from the same IP
+grouping should be permitted.
+
+For IP grouping:
+
+* **IPv6 addresses** are grouped by ASN.
+* **IPv4 addresses** are grouped by `/16` prefixes, except for [legacy Class A
+blocks](https://en.wikipedia.org/wiki/List_of_assigned_/8_IPv4_address_blocks),
+which are grouped by `/8` prefixes.
+
+Since a single node can advertise multiple addresses, a peer MUST NOT be added
+to the routing table if any of its addresses already exceed the allowed
+representation within the table.
+
+### Routing Table Refresh
```

Comment:
This section seems general with the exception of any numbers being recommended.

Reply:
https://github.com/ipfs/specs/pull/497#discussion_r2016997925